### ISSUE TITLE
create-vue fails in Linux Mint v22

### ISSUE THREAD

**Issue by mdr0414** [vuejs/create-vue#583]
Simply trying to scaffold a new project using "npm create vue@latest" and I get terminal output:

> npx
> create-vue

sh: 1: create-vue: not found
npm error code 127
npm error path /home/mark/Programming/Vue/supabase-ssr
npm error command failed
npm error command sh -c create-vue
npm error A complete log of this run can be found in: /home/mark/.npm/_logs/2024-10-16T12_20_20_375Z-debug-0.log

Here is the log:
0 verbose cli /home/mark/.nvm/versions/node/v22.6.0/bin/node /usr/local/bin/npm
1 info using npm@10.9.0
2 info using node@v22.6.0
3 silly config load:file:/usr/local/lib/node_modules/npm/npmrc
4 silly config load:file:/home/mark/Programming/Vue/supabase-ssr/.npmrc
5 silly config load:file:/home/mark/.npmrc
6 silly config load:file:/etc/npmrc
7 verbose title npm create vue@latest
8 verbose argv "create" "vue@latest"
9 verbose logfile logs-max:10 dir:/home/mark/.npm/_logs/2024-10-16T12_20_20_375Z-
10 verbose logfile /home/mark/.npm/_logs/2024-10-16T12_20_20_375Z-debug-0.log
11 silly logfile start cleaning logs, removing 1 files
12 silly logfile done cleaning log files
13 silly packumentCache heap:4345298944 maxSize:1086324736 maxEntrySize:543162368
14 http fetch GET 200 https://registry.npmjs.org/create-vue 435ms (cache miss)
15 silly packumentCache heap:4345298944 maxSize:1086324736 maxEntrySize:543162368
16 verbose stack Error: command failed
16 verbose stack     at promiseSpawn (/usr/local/lib/node_modules/npm/node_modules/@npmcli/promise-spawn/lib/index.js:22:22)
16 verbose stack     at spawnWithShell (/usr/local/lib/node_modules/npm/node_modules/@npmcli/promise-spawn/lib/index.js:124:10)
16 verbose stack     at promiseSpawn (/usr/local/lib/node_modules/npm/node_modules/@npmcli/promise-spawn/lib/index.js:12:12)
16 verbose stack     at runScriptPkg (/usr/local/lib/node_modules/npm/node_modules/@npmcli/run-script/lib/run-script-pkg.js:77:13)
16 verbose stack     at runScript (/usr/local/lib/node_modules/npm/node_modules/@npmcli/run-script/lib/run-script.js:9:12)
16 verbose stack     at run (/usr/local/lib/node_modules/npm/node_modules/libnpmexec/lib/run-script.js:48:10)
16 verbose stack     at async exec (/usr/local/lib/node_modules/npm/node_modules/libnpmexec/lib/index.js:298:10)
16 verbose stack     at async Init.execCreate (/usr/local/lib/node_modules/npm/lib/commands/init.js:135:5)
16 verbose stack     at async Init.exec (/usr/local/lib/node_modules/npm/lib/commands/init.js:44:14)
16 verbose stack     at async Npm.exec (/usr/local/lib/node_modules/npm/lib/npm.js:207:9)
17 error code 127
18 error path /home/mark/Programming/Vue/supabase-ssr
19 error command failed
20 error command sh -c create-vue
21 verbose cwd /home/mark/Programming/Vue/supabase-ssr
22 verbose os Linux 6.8.0-45-generic
23 verbose node v22.6.0
24 verbose npm  v10.9.0
25 verbose exit 127
26 verbose code 127
27 error A complete log of this run can be found in: /home/mark/.npm/_logs/2024-10-16T12_20_20_375Z-debug-0.log

Any ideas what could be wrong?  I've cleared npm cache as well.
tia
_mark

**Comment by Yordan-Ramchev** [comment]
Hey @mdr0414, I've tested it on macOS with node 22, and it's working as expected.

<img width="718" alt="Screenshot 2024-10-16 at 16 03 10" src="https://github.com/user-attachments/assets/99292330-1838-4a62-b3bf-223a0bf5fa06">

Based on your **Log**:

> 3 silly config load:file:/usr/local/lib/node_modules/npm/npmrc
4 silly config load:file:/home/mark/Programming/Vue/supabase-ssr/.npmrc
5 silly config load:file:/home/mark/.npmrc
6 silly config load:file:/etc/npmrc

I assume that there is something wrong with **.npmrc** config on your machine, because I had a similar issue with NPM. 

I've fixed it by deleting the registry:
`npm config delete registry`

**Comment by mdr0414** [comment]
hmm ... nope, that did not work. Same problem ...
> npx
> create-vue

sh: 1: create-vue: not found
npm error code 127

**Comment by cexbrayat** [comment]
@mdr0414 Can you try installing a more recent node version (22.10 for example) and check if that solve your issue?

**Comment by mdr0414** [comment]
updated node to v22.10.0 and still get the same error:
> npx
> create-vue

sh: 1: create-vue: not found
npm error code 127
npm error path /media/mark/T7/Test
npm error command failed
npm error command sh -c create-vue
npm error A complete log of this run can be found in: /home/mark/.npm/_logs/2024-10-19T13_12_52_625Z-debug-0.log

14 http fetch GET 200 https://registry.npmjs.org/create-vue 690ms (cache miss)
15 silly packumentCache heap:4345298944 maxSize:1086324736 maxEntrySize:543162368
16 verbose stack Error: command failed
16 verbose stack     at promiseSpawn (/home/mark/.nvm/versions/node/v22.10.0/lib/node_modules/npm/node_modules/@npmcli/promise-spawn/lib/index.js:22:22)
16 verbose stack     at spawnWithShell (/home/mark/.nvm/versions/node/v22.10.0/lib/node_modules/npm/node_modules/@npmcli/promise-spawn/lib/index.js:124:10)
16 verbose stack     at promiseSpawn (/home/mark/.nvm/versions/node/v22.10.0/lib/node_modules/npm/node_modules/@npmcli/promise-spawn/lib/index.js:12:12)
16 verbose stack     at runScriptPkg (/home/mark/.nvm/versions/node/v22.10.0/lib/node_modules/npm/node_modules/@npmcli/run-script/lib/run-script-pkg.js:77:13)
16 verbose stack     at runScript (/home/mark/.nvm/versions/node/v22.10.0/lib/node_modules/npm/node_modules/@npmcli/run-script/lib/run-script.js:9:12)
16 verbose stack     at run (/home/mark/.nvm/versions/node/v22.10.0/lib/node_modules/npm/node_modules/libnpmexec/lib/run-script.js:48:10)
16 verbose stack     at async exec (/home/mark/.nvm/versions/node/v22.10.0/lib/node_modules/npm/node_modules/libnpmexec/lib/index.js:298:10)
16 verbose stack     at async Init.execCreate (/home/mark/.nvm/versions/node/v22.10.0/lib/node_modules/npm/lib/commands/init.js:135:5)
16 verbose stack     at async Init.exec (/home/mark/.nvm/versions/node/v22.10.0/lib/node_modules/npm/lib/commands/init.js:44:14)
16 verbose stack     at async Npm.exec (/home/mark/.nvm/versions/node/v22.10.0/lib/node_modules/npm/lib/npm.js:207:9)
17 error code 127
18 error path /media/mark/T7/Test
19 error command failed
20 error command sh -c create-vue
21 verbose cwd /media/mark/T7/Test
22 verbose os Linux 6.8.0-47-generic
23 verbose node v22.10.0
24 verbose npm  v10.9.0
25 verbose exit 127
26 verbose code 127

At a total loss as to what to do.  I get a similar error when trying to create a new Vite project with: npm create vite@latest ...
sh: 1: create-vite: not found

**Comment by haoqunjiang** [comment]
I suspect it has something to do with your nvm installation/configuration.
Because your node path points to a binary installed by nvm (`/home/mark/.nvm/versions/node/v22.6.0/bin/node`), while npm points to a global binary: `/usr/local/bin/npm`

**Comment by mdr0414** [comment]
thanks for the reply ... so do you think uninstalling npm (and nvm?) and reinstalling should solve it?  And, does uninstalling nvm automatically uninstall npm? or vise versa...

**Comment by haoqunjiang** [comment]
I don't know… as I don't use nvm myself.
Maybe you can follow the steps in their official docs? https://github.com/nvm-sh/nvm#uninstalling--removal

**Comment by mdr0414** [comment]
Ok -- just to close this up as reference for others ... 
I removed nvm.  Removed node, purged any related files, and re-installed node (20).  All works as it should now.  
As reference: for some reason, re-running "npm install" after deleting package-lock.json and the node_modules folder to rebuild everything, would not create the node_modules/.bin folder or node_modules/.vite folder.  I was "testing" all this on an external ssd drive formatted as exfat and somehow all is related to symlinks, which are not supported on exfat partitions.  After I FINALLY got that error fiddling around, I used a home folder to create a new project and all ran fine (again, after reinstalling nodejs).  Thanks for all the suggestions ...
